### PR TITLE
Add support for float read_pixels

### PR DESCRIPTION
**Pull request by MortimerGoro**
This change is [Reviewable](https://reviewable.io/reviews/servo/gleam/127)

### CODE REVIEW

File: src/gl.rs [hunk 1]
```diff
@@ -54,6 +54,7 @@ fn calculate_length(width: GLsizei, height: GLsizei, format: GLenum, pixel_type:
     };
     let depth = match pixel_type {
         ffi::UNSIGNED_BYTE => 1,
+        ffi::FLOAT=> 4,
```
Comment (kvark) [inline src/gl.rs]: shouldn't we also be checking `format` for `ffi::DEPTH`?

Comment (MortimerGoro) [reply src/gl.rs]: why? ```depth = match pixel_type```only gets the bytes per color component. The ```match format``` gets the number of color components.

Comment (kvark) [reply src/gl.rs]: What do you pass when reading the depth? I'd expect `format == ffi::DEPTH` and `pixel_type == ffi::FLOAT`

Comment (MortimerGoro) [reply src/gl.rs]: ok, I added the DEPTH format check. According to the spec glReadPixels uses ffi::DEPTH_COMPONENT instead of ffi::DEPTH

**Comment by kvark** [comment]
Thank you!
@bors-servo r+

**Comment by MortimerGoro** [comment]
Could you publish gleam 4.9 to crates.io?

**Comment by kvark** [comment]
@MortimerGoro I will, once both PRs land ;)

**Comment by kvark** [comment]
https://crates.io/crates/gleam/0.4.9

🤞 for not breaking anything

**Comment by MortimerGoro** [comment]
@kvark Thanks!

**Pull request merged**